server/middlewares/checkPermissions.js

```diff
@@ -0,0 +1,39 @@
+import permissionsConfig from '../config/permissions.config';
+import { badHttpResponse } from '../utilities/httpResponse';
+import processUrl from '../utilities/processUrl';
+import userRepo from '../repository/userRepository';
+
+/**
+ * Middleware regulating access control on protected routes
+ * @param {object} request Request Object
+ * @param {Object} response Response Object
+ * @param {function} next Next middleware
+ * @returns {function | object} Next function on the middleware chain or an error object
+ */
+const checkPermissions = async (request, response, next) => {
+  const resource = processUrl(request.url);
+
+  const user = await userRepo.getUserByParam('id', request.userId);
+  if (!user) {
+    return badHttpResponse(response, 403, 'ACCESS DENIED');
+  }
+
+  const { email, role } = user.dataValues;
+
+  if (role === 'admin' && email === process.env.SUPER_ADMIN_EMAIL) {
```

Comment:
why isnt this just `role === 'admin'`

Reply:
We decided that a user could be upgraded to an admin but this doesnt give him superadmin privileges like deleting other admins or modifying the role of a superadmin. I am making a check for the superadmin (using the email is as good as using any other primary identifier, seeing as emails are unique). On instance of a superadmin, then we dont care what he was trying to do. He gets automatic access.